### PR TITLE
Add YouTube ad-blocking omnibar badge

### PR DESCRIPTION
**Pull request by CrisBarreiro**
Task/Issue URL: https://app.asana.com/1/137249556945/project/72649045549333/task/1214489465023871?focus=true
Tech Design URL (if applicable): 

### Description
* Add a new omnibar animation for YouTube ad blocking

### Steps to test this PR

_Feature 1_
- [x] Fresh install from develop
- [x] Smoke test trackers and cookies animation

_Feature 2_
- [x] Go to feature flag inventory and enable `adBlockingExtension` along with sub-setting`adBlockingUXImprovements`
- [x] Go to settings -> Ad Blocking and enable the feature
- [ ] Load a YouTube video. Check the new animation shows (no cookies or trackers show)
- [ ] From there, navigate to another video (SPA navigation), check the animation shows again (no cookies or trackers show)
- [ ] Refresh the page, check the animation shows again (no cookies or trackers show)
- [ ] Click on the YouTube logo from the header (SPA to YT home)
- [ ] Check no ad blocking animation is displayed 
- [ ] Smoke test trackers and cookies animations on other sites 

### UI changes
| Before  | After |
| ------ | ----- |
|n/a|image|

---

> [!NOTE]
> **Medium Risk**
> Changes address-bar animation orchestration and navigation timing (deferred badges, exclusivity), which is user-visible and race-prone, though heavily unit-tested and gated by ad-blocking inject eligibility.
> 
> **Overview**
> Adds an **omnibar badge animation** when YouTube ad blocking is active on supported video URLs (`watch`, `shorts`, `live`, `clip`), wired through a new `AdBlockingOmnibarAnimationProvider` that returns **Show**, **Skip**, or **Retain** depending on URL, feature state, and whether the navigation is a full load vs in-page SPA change.
> 
> The browser tab layer consults that provider on `pageChanged` and SPA `urlUpdated`, defers showing the badge until load reaches 100% when appropriate, and uses a **navigation token** so stale deferred badges are not flushed after the user navigates away. While the badge is claimed, **tracker and autoconsent/cookie address-bar animations are suppressed**; focus or custom-tab mode can suppress the UI and release the claim via `onAdBlockingAnimationSuppressed`.
> 
> Omnibar UI gains a cookie-style slide animation path (`LaunchAdBlockingAnimation`, dedicated views/scenes, icon and localized `ad_blocking_omnibar_badge_text`), with unit tests covering URL logic and browser/orchestration edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 85e734a8d51e408911df6a90335c94dcfdaa31fb. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by CrisBarreiro** [comment]
* **#9155** <a href="https://app.graphite.com/github/pr/duckduckgo/Android/9155?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/duckduckgo/Android/9155?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `develop`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt [hunk 1]
```diff
@@ -2060,6 +2069,8 @@ class BrowserTabViewModel @Inject constructor(
 
             is WebNavigationStateChange.PageCleared -> pageCleared()
             is WebNavigationStateChange.UrlUpdated -> {
+                // Drop any prior page's deferred badge synchronously (see NewPage above).
+                resetAdBlockingAnimationState()
```
[the remark this answers is not in the document]
Comment (CrisBarreiro) [reply app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt]: This is expected, otherwise, we can end up showing the badge on the wrong page

File: ad-blocking/ad-blocking-impl/src/main/res/values/donottranslate.xml [hunk 2]
```diff
@@ -15,4 +15,5 @@
   -->
 
 <resources>
+    <string name="ad_blocking_omnibar_badge_text">YouTube Ad Block On</string>
```
Comment (malmstein) [inline ad-blocking/ad-blocking-impl/src/main/res/values/donottranslate.xml]: nit: "YouTube Ad Block On" is user-facing but lives in donottranslate, so it won't localize. fine while the feature is rolling out, worth moving to a translatable strings file before wide release.

File: app/src/main/java/com/duckduckgo/app/browser/omnibar/animations/addressbar/BrowserLottieTrackersAnimatorHelper.kt [hunk 3]
```diff
@@ -191,6 +202,116 @@ class BrowserLottieTrackersAnimatorHelper @Inject constructor(
...
+        startAdBlockingAnimation(context, omnibarViews + shieldViews, icon, text)
+    }
+
+    private fun startAdBlockingAnimation(
+        context: Context,
+        omnibarViews: List<View>,
+        iconRes: Int,
+        textRes: Int,
+    ) {
+        if (omnibarViews.any { it.id == R.id.customTabDomain }) return // not shown in custom tabs
```
Comment (malmstein) [inline app/src/main/java/com/duckduckgo/app/browser/omnibar/animations/addressbar/BrowserLottieTrackersAnimatorHelper.kt:233]: in a custom tab this early returns without releasing `adBlockingAnimationClaimed`, so trackers stay suppressed and no badge shows. should we route custom tabs through the same suppressed signal as the focused case, so the claim is released when the badge can't render? niche and self-heals on the next Skip, not blocking.

**Review by malmstein — APPROVED** [review]
Installed on device. Clean, well-structured feature: `AdBlockingOmnibarAnimationProvider` (api) with a Show/Skip/Retain sealed result, video-id-keyed dedup in `RealAdBlockingOmnibarAnimation`, and an exclusivity model where the ad-blocking badge suppresses the tracker/cookie animation (pixel still fires). The deferred-badge machinery (pending badge + nav token, flushed at progress 100) correctly avoids a stale badge across navigations, and the test matrix is thorough (deferred, stale-token, rapid SPA, Retain, focus-suppressed, custom-tab guard). Two non-blocking items inline.

File: ad-blocking/ad-blocking-impl/src/main/res/values/donottranslate.xml (hunk 2 above)
Comment (CrisBarreiro) [reply ad-blocking/ad-blocking-impl/src/main/res/values/donottranslate.xml]: Yes, I have translations ready on another branch, just didn't want to push them yet

File: app/src/main/java/com/duckduckgo/app/browser/omnibar/animations/addressbar/BrowserLottieTrackersAnimatorHelper.kt (hunk 3 above)
Comment (CrisBarreiro) [reply app/src/main/java/com/duckduckgo/app/browser/omnibar/animations/addressbar/BrowserLottieTrackersAnimatorHelper.kt:233]: Fixed
